### ISSUE TITLE
resolveUrl missing from 2.x

### ISSUE THREAD

**Issue by arthurevans** [Polymer/polymer#3992]
Doesn't appear to be in the prototype for 2.x. The omission isn't noted in the current release notes.

**Comment by arthurevans** [comment]
Hm. It appears that this _does_ exist if I extend LegacyElement, but it's not clear that it's working correctly.

Running PSK in 1.x `this.resolveUrl('test')` from the main app element => 'http://localhost:9000/src/test'

Using 2.x with LegacyElement => 'test'

**Comment by arthurevans** [comment]
Currently both PSK and Shop use this to resolve import URLs for lazy-loaded pages, so this is a blocker.

**Comment by sorvell** [comment]
We're inclined to move this to `Polymer.Element`. It goes together with `dom-module` which `Polymer.Element` also uses. 

The platform answer would be something like this:

```
<script> /* executing inside htmlimport */
const base = document.currentScript.ownerDocument.baseURI;
let urlRelativeToImport = new URL('foo.html', base);
// ...
```

This doesn't work after bundling and `resolveUrl` accounts for this by coordinating via `dom-module`.

**Comment by kevinpschaaf** [comment]
The reason `resolveUrl` didn't work even on `class extend LegacyElement` is because in the class-based syntax the static get is() getter is only on the constructor, and never gets put on the prototype/instance (and `resolveUrl` only looked on the instance, which is ok with the legacy API since `is` does end up on the prototype there).  Resolved in #4090.

**Comment by kevinpschaaf** [comment]
Resolved via https://github.com/Polymer/polymer/pull/4090
